ext.autodoc: fix formatting instance annotations

joostvanzwieten opened the pull request:
Currently `format_annotation` fails on instances, because instances don't have `__module__` and `__qualname__` attributes.  Defer building the `qualified_name` of an annotation until we have established that the annotation is a type.

nijel committed:
Support any encoding in HEAD requests

Completely support would be more problematic and this already covers
most usual case.

Fixes #2775

Signed-off-by: Michal Čihař <michal@cihar.com>

tk0miya committed:
Merge pull request #2809 from nijel/stable

Support gzip encoding in HEAD requests

tk0miya committed:
Add Seaborn project to EXAMPLES (ref: #2814)

joostvanzwieten committed:
ext.autodoc: fix formatting instance annotations

Currently `format_annotation` fails on instances, because instances don't have
`__module__` and `__qualname__` attributes.  Defer building the
`qualified_name` of an annotation until we have established that the annotation
is a type.

EricFromCanada committed:
remove extra colons in C++ domain examples

tk0miya committed:
Merge pull request #2824 from EricFromCanada/stable

remove extra colons in C++ domain examples

tk0miya commented:
LGTM
It would be nice if you rebase this to stable branch.

In addition, could you add an entry to CHANGES file?

joostvanzwieten commented:
I've rebased the commit onto stable and added myself to `AUTHORS`. I'm not sure though how to change the pull request to point to stable instead of master. Do you want me to open another pull request?

joostvanzwieten commented:
Sorry, misread your comment. I've updated `CHANGES` instead of `AUTHORS`.

tk0miya commented:
Yes, please open new pull request for stable.
Thanks,

joostvanzwieten commented:
Superseded by #2833 (different target branch).
